src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -180,6 +181,32 @@ internal partial class Binder
             return Tuple.Create(refKinds, types, names, isAsync);
         }
 
+        private void CheckParanthesizedLambdaParameters(
+            SeparatedSyntaxList<ParameterSyntax> parameterSyntaxList, DiagnosticBag diagnostics)
+        {
+            if (parameterSyntaxList.Count > 0)
+            {
+                var hasTypes = parameterSyntaxList[0].Type != null;
```

Comment:
I understand this preserves the previous behavior, but it raises the question: should we check all the parameters for presence of type?
(if yes, let's file an issue)